Dockerfile

```diff
@@ -2,16 +2,18 @@ FROM python:3.6.8-alpine3.9@sha256:c3b0a0317d0759a0c0b8cf2b2c60ea09729bc9b7faf46
 
 ARG requirements=requirements.txt
 
-RUN apk add --no-cache postgresql-libs bash && \
-    apk add --no-cache --virtual .build-deps gcc musl-dev postgresql-dev && \
+RUN apk add --no-cache bash && \
     wget -q https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -P /usr/local/bin && \
-    chmod +x /usr/local/bin/wait-for-it.sh
+    chmod +x /usr/local/bin/wait-for-it.sh && \
+    if [ "$requirements" = "requirements-dev.txt" ]; then apk add --no-cache postgresql-libs && \
+    apk add --no-cache --virtual .build-deps gcc musl-dev postgresql-dev; fi
 
 WORKDIR /app
 
 COPY requirements.txt requirements-dev.txt /app/
 RUN pip install --no-cache-dir -r ${requirements} && \
-    apk --purge del .build-deps
+    if [ "$requirements" = "requirements-dev.txt" ]; then apk --purge del .build-deps; fi
```

Comment:
If we're optimizing for image size, shouldn't all this be done in a single `RUN` line? Otherwise the layers would still contain the stuff between this `RUN` and the previous one..

Reply:
This **is** a single `RUN` line:

```dockerfile
RUN apk add --no-cache bash && \
    wget -q https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh -P /usr/local/bin && \
    chmod +x /usr/local/bin/wait-for-it.sh && \
    if [ "$requirements" = "requirements-dev.txt" ]; then apk add --no-cache postgresql-libs && \
    apk add --no-cache --virtual .build-deps gcc musl-dev postgresql-dev; fi
```

Or did I misunderstand your comment?

Reply:
No I meant we should consider merging this `RUN` line with the one on lines 6..9. Would require moving/reorganizing the stuff inbetween though, so not sure if it's worth ti

Reply:
You are absolutely right! Please excuse my confusion... :)